rohmu/object_storage/local.py

```diff
@@ -53,6 +59,7 @@ def __init__(
     ) -> None:
         prefix = os.path.join(directory, (prefix or "").strip("/"))
         super().__init__(prefix=prefix, notifier=notifier, statsd_info=statsd_info)
+        self._concurrent_uploads: dict[str, tuple[ConcurrentUploadData, Optional[Metadata], dict[int, str]]] = {}
```

Comment:
maybe we can move this into the base class with `_get_upload_info` since most implementations would use something like this to avoid parsing the upload_id everytime...